### PR TITLE
Introduce OVIS's log library

### PR DESCRIPTION
**Pull request by nichamon**

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -125,6 +127,7 @@ OPTION_DEFAULT_ENABLE([mmalloc], [ENABLE_MMALLOC])
 OPTION_DEFAULT_ENABLE([ovis_ctrl], [ENABLE_OVIS_CTRL])
 OPTION_DEFAULT_ENABLE([ovis_util], [ENABLE_OVIS_UTIL])
 OPTION_DEFAULT_ENABLE([ovis_auth], [ENABLE_OVIS_AUTH])
+OPTION_DEFAULT_ENABLE([ovis_log], [ENABLE_OVIS_LOG])
```
[the remark this answers is not in the document]
Comment (nichamon) [reply configure.ac:130]: I remember that we have the option to disable any libraries in lib to make an image of a specific library or set of libraries. I put the option in to be consistent with the other libraries.

**Comment by tom95858** [comment]
@nichamon, I do not understand the purpose of all of this configuration logic. Correct me if I'm wrong, but if we disable OVIS_UTIL, for example, the code won't build. This will also be true of the log library when it is merged. I am suggesting that all of this configuration logic is the result of "foolish consistency," but no reasonable purpose. Perhaps I'm missing some reasonable use case.

**Comment by nichamon** [comment]
@tom95858 I agree that it doesn't make sense to have the disabling option for ovis_log. I forgot that we want to be able to use it anywhere in the code, including the libraries in lib. I was only thinking about LDMSD, so it made sense at the time to disable ovis_log when, for example, we want only to build Zap. I will remove the option for ovis_log.

**Comment by tom95858** [comment]
@nichamon, @morrone I agree and perhaps we should look at some of this other configuration logic (in a different pull request) that is never really utilized. I just wanted to be sure I wasn't missing something.

**Comment by nichamon** [comment]
@tom95858 I removed the option for `ovis_log` and `ovis_ev` in this patch. The pull request depends on #1108 that removes the option of coll and ovis_util.

**Pull request merged**